testbed/tests/app/test_desktop.py

```diff
@@ -282,6 +328,209 @@ async def test_full_screen(app, app_probe):
         window2.close()
 
 
+async def test_presentation_mode_with_main_window(
+    app, app_probe, main_window, main_window_probe
+):
+    """The app can enter presentation mode with just the main window."""
+    # This test is required since the toolbar is present only on the main window.
+    try:
+        main_window.toolbar.add(app.cmd1)
+        extra_window = toga.Window(
+            "Extra Test Window", position=(150, 150), size=(200, 200)
+        )
+        extra_window.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
+        extra_window.show()
+        extra_window_probe = window_probe(app, extra_window)
+        # Add delay for gtk to show the windows
+        await app_probe.redraw("Extra window is visible", delay=0.5)
+
+        main_window_initial_content_size = main_window_probe.presentation_content_size
+        extra_window_initial_content_size = extra_window_probe.presentation_content_size
+        # Enter presentation mode with only main window
+        app.enter_presentation_mode([main_window])
+        # Add delay for gtk to show the windows
+        await app_probe.redraw("App is in presentation mode", delay=0.5)
+        assert app.is_presentation_mode
+
+        # Extra  window should not be in presentation mode.
+        assert not extra_window_probe.is_window_state(
+            WindowState.PRESENTATION
+        ), "Extra Window:"
+        assert (
+            extra_window_probe.presentation_content_size
+            == extra_window_initial_content_size
+        ), "Extra Window"
+
+        # Main Window should be in presentation mode.
+        assert main_window_probe.is_window_state(WindowState.PRESENTATION)
+        assert main_window_probe.presentation_content_size[0] > 1000
+        assert main_window_probe.presentation_content_size[1] > 700
+
+        # Exit presentation mode
+        app.exit_presentation_mode()
+        await app_probe.redraw("App is no longer in presentation mode", delay=0.5)
+        assert not app.is_presentation_mode
+
+        assert (
+            main_window_probe.presentation_content_size
+            == main_window_initial_content_size
+        )
+
+    finally:
+        main_window.toolbar.clear()
+        extra_window.close()
+
+
+async def test_presentation_mode_with_screen_window_dict(app, app_probe):
```

Comment:
We don't need to exercise the *interface* - just the implementation. There's no difference to the backend API whether the user specifies presentation mode as a dict or a list.

Reply:
We *do*, however, need to test all the edge cases that can exist when you move between window states - for example, moving a window from presentation mode to minimised mode.

Reply:
I have added new tests.